server/controllers/chartsController/index.js

```diff
@@ -116,11 +117,17 @@ const getAllSubjects = async ({ dataDb, chart, userAccess, filters }) => {
     )
 
     for await (let doc of criteriaCursor) {
-      const data = await dataDb
-        .collection(doc.collection)
-        .findOne({ $or: mongoFilterQuery })
+      const data = await dataDb.collection(doc.collection).aggregate(aggregate)
+      const subjectCriteriaData = await data.next()
```

Comment:
what does `data.next()` do?

Reply:
The aggregate method returns a cursor, but to access the cursor data I'm using next(). There are other ways to convert and access data like toArray() etc...